### PR TITLE
Create Shifts policy commandlets

### PR DESCRIPTION
**Pull request by akumar39**

### CODE REVIEW

**Review by get-itips — COMMENTED** [review]
Hello Aarthi, how are you?
Looks good,
Are these cmdlets going to be released soon on the Teams GA or Public Preview version? which version exactly?

so we can update this later https://docs.microsoft.com/en-us/microsoftteams/teams-powershell-release-notes

File: teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+## SYNTAX
+
+```
+New-CsTeamsShiftsPolicy [-Identity] <XdsIdentity> [-EnableShiftPresence <Boolean>]
+ [-ShiftNoticeFrequency <String>] [-ShiftNoticeMessageType <String>] [-ShiftNoticeMessageCustom <String>]
+ [-AccessType <String>] [-AccessGracePeriodMinutes <Int64>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+This cmdlet allows you to create a TeamsShiftPolicy instance. Use this to also set the policy name, user's shift based presence(EnableShiftPresence) and Teams off shift warning message-specific settings(ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).
```
Comment (dariomws) [inline teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md:23]:
```suggestion
This cmdlet allows you to create a TeamsShiftPolicy instance. Use this to also set the policy name, user's shift based presence (EnableShiftPresence) and Teams off shift warning message-specific settings (ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).

```

File: teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AccessType
+Indicates the Teams access type granted to the user. Today, only unrestricted access to Teams app is supported.
+Use 'UnrestrictedAccess_TeamsApp' as the value for this setting, or is set by default.
+For Teams Off Shift Access Control, the option to show the user a blocking dialog message is supported. Once user accepts this message, it is audit logged and user has usual access to Teams. Set other off shift warning message-specific settings to configure off shift access controls for the user.
```
Comment (dariomws) [inline teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md:61]:
```suggestion
Indicates the Teams access type granted to the user. Today, only unrestricted access to Teams app is supported.
Use 'UnrestrictedAccess_TeamsApp' as the value for this setting, or is set by default.
For Teams Off Shift Access Control, the option to show the user a blocking dialog message is supported. Once the user accepts this message, it is audit logged and the user has usual access to Teams. Set other off shift warning message-specific settings to configure off shift access controls for the user.

```

File: teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+'Message3' - You won't be compensated for time using Teams. By accepting, you acknowledge that you understand the statement above.
+'Message4' - You're not authorized to use Teams while off shift. By accepting, you acknowledge your use of Teams is against your employer's policy.
+'Message5' - Access to Teams is turned off during non-working hours. You will be able to access the app when your next shift starts.
+'Message6' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. Access to corporate resources are only allowed during approved working hours and should be recorded as hours worked in your employer’s timekeeping system.
+'Message7' - Your employer has turned off access to Teams during non-working hours. Refer to your employer's guidelines on using this app outside working hours.
+'DefaultMessage' - You aren't authorized to use Microsoft Teams during non-working hours and will only be compensated for using it during approved working hours.
+'CustomMessage'
```
Comment (dariomws) [inline teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md:115]:
```suggestion
The warning message is shown in the blocking dialog when a user access Teams off shift hours. Select one of 7 Microsoft provided messages, a default message or a custom message. 
'Message1' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. By accepting, you acknowledge that your use of Teams while off shift is not authorized and you will not be compensated.
'Message2' - Accessing this app outside working hours is voluntary. You won't be compensated for time spent on Teams. Refer to your employer's guidelines on using this app outside working hours. By accepting, you acknowledge that you understand the statement above.
'Message3' - You won't be compensated for time using Teams. By accepting, you acknowledge that you understand the statement above.
'Message4' - You're not authorized to use Teams while off shift. By accepting, you acknowledge your use of Teams is against your employer's policy.
'Message5' - Access to Teams is turned off during non-working hours. You will be able to access the app when your next shift starts.
'Message6' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. Access to corporate resources are only allowed during approved working hours and should be recorded as hours worked in your employer’s timekeeping system.
'Message7' - Your employer has turned off access to Teams during non-working hours. Refer to your employer's guidelines on using this app outside working hours.
'DefaultMessage' - You aren't authorized to use Microsoft Teams during non-working hours and will only be compensated for using it during approved working hours.
'CustomMessage'

```

File: teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+Applicable: Microsoft Teams
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (dariomws) [inline teams/teams-ps/teams/New-CsTeamsShiftsPolicy.md:160]:
```suggestion
This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

```

File: teams/teams-ps/teams/Remove-CsTeamsShiftsPolicy.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Applicable: Microsoft Teams
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (dariomws) [inline teams/teams-ps/teams/Remove-CsTeamsShiftsPolicy.md:49]:
```suggestion
This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

```

File: teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+This cmdlet allows you to set or update properties of a TeamsShiftPolicy instance, including user's shift based presence and Teams off shift warning message-specific settings.
+
+## SYNTAX
+
+```
+Set-CsTeamsShiftsPolicy [[-Identity] <XdsIdentity>] [-EnableShiftPresence <Boolean>] [-ShiftNoticeFrequency <String>] [-ShiftNoticeMessageType <String>] [-ShiftNoticeMessageCustom <String>] [-AccessType <String>] [-AccessGracePeriodMinutes <Int64>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+This cmdlet allows you to set or update properties of a TeamsShiftPolicy instance. Use this to set the policy name, user's shift based presence(EnableShiftPresence) and Teams off shift warning message-specific settings(ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).
```
Comment (dariomws) [inline teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md:21]:
```suggestion
This cmdlet allows you to set or update properties of a TeamsShiftPolicy instance. Use this to set the policy name, user's shift based presence (EnableShiftPresence) and Teams off shift warning message-specific settings (ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).

```

File: teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AccessType
+Indicates the Teams access type granted to the user. Today, only unrestricted access to Teams app is supported.
+Use 'UnrestrictedAccess_TeamsApp' as the value for this setting, or is set by default.
+For Teams Off Shift Access Control, the option to show the user a blocking dialog message is supported. Once user accepts this message, it is audit logged and user has usual access to Teams. Set other off shift warning message-specific settings to configure off shift access controls for the user.
```
Comment (dariomws) [inline teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md:51]:
```suggestion
Indicates the Teams access type granted to the user. Today, only unrestricted access to Teams app is supported.
Use 'UnrestrictedAccess_TeamsApp' as the value for this setting, or is set by default.
For Teams Off Shift Access Control, the option to show the user a blocking dialog message is supported. Once the user accepts this message, it is audit logged and the user has usual access to Teams. Set other off shift warning message-specific settings to configure off shift access controls for the user.

```

File: teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+'Message3' - You won't be compensated for time using Teams. By accepting, you acknowledge that you understand the statement above.
+'Message4' - You're not authorized to use Teams while off shift. By accepting, you acknowledge your use of Teams is against your employer's policy.
+'Message5' - Access to Teams is turned off during non-working hours. You will be able to access the app when your next shift starts.
+'Message6' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. Access to corporate resources are only allowed during approved working hours and should be recorded as hours worked in your employer’s timekeeping system.
+'Message7' - Your employer has turned off access to Teams during non-working hours. Refer to your employer's guidelines on using this app outside working hours.
+'DefaultMessage' - You aren't authorized to use Microsoft Teams during non-working hours and will only be compensated for using it during approved working hours.
+'CustomMessage'
```
Comment (dariomws) [inline teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md:90]:
```suggestion
The warning message is shown in the blocking dialog when a user access Teams off shift hours. Select one of 7 Microsoft provided messages, a default message or a custom message. 
'Message1' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. By accepting, you acknowledge that your use of Teams while off shift is not authorized and you will not be compensated.
'Message2' - Accessing this app outside working hours is voluntary. You won't be compensated for time spent on Teams. Refer to your employer's guidelines on using this app outside working hours. By accepting, you acknowledge that you understand the statement above.
'Message3' - You won't be compensated for time using Teams. By accepting, you acknowledge that you understand the statement above.
'Message4' - You're not authorized to use Teams while off shift. By accepting, you acknowledge your use of Teams is against your employer's policy.
'Message5' - Access to Teams is turned off during non-working hours. You will be able to access the app when your next shift starts.
'Message6' - Your employer does not authorize or approve of the use of its network, applications, systems, or tools by non-exempt or hourly employees during their non-working hours. Access to corporate resources are only allowed during approved working hours and should be recorded as hours worked in your employer’s timekeeping system.
'Message7' - Your employer has turned off access to Teams during non-working hours. Refer to your employer's guidelines on using this app outside working hours.
'DefaultMessage' - You aren't authorized to use Microsoft Teams during non-working hours and will only be compensated for using it during approved working hours.
'CustomMessage'

```

File: teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+Applicable: Microsoft Teams
+Required: False
+Position: Named
+Default value: Always
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AccessGracePeriodMinutes
+Indicates the grace period time in minutes between when the first shift starts or last shift ends and when access is blocked.
```
Comment (dariomws) [inline teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md:135]:
```suggestion
Indicates the grace period time in minutes between when the first shift starts, or last shift ends and when access is blocked.

```

File: teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+Applicable: Microsoft Teams
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (dariomws) [inline teams/teams-ps/teams/Set-CsTeamsShiftsPolicy.md:150]:
```suggestion
This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

```

File: teams/teams-ps/teams/Get-CsTeamsShiftsPolicy.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+## SYNTAX
+
+### Identity (Default)
+```
+Get-CsTeamsShiftsPolicy [[-Identity] <XdsIdentity>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+This cmdlet allows you to get properties of a TeamsShiftPolicy instance. Use this to get the policy name, user's shift based presence(EnableShiftPresence) and Teams off shift warning message-specific settings(ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).
```
Comment (dariomws) [inline teams/teams-ps/teams/Get-CsTeamsShiftsPolicy.md:22]:
```suggestion
This cmdlet allows you to get properties of a TeamsShiftPolicy instance. Use this to get the policy name, user's shift based presence (EnableShiftPresence) and Teams off shift warning message-specific settings (ShiftNoticeMessageType, ShiftNoticeMessageCustom, ShiftNoticeFrequency, AccessGracePeriodMinutes).

```

**Review by dariomws — APPROVED** [review]
I only recommend a few details.

Also, I adhere to what @ get-itips says.

Thanks!

**Comment by akumar39** [comment]
> Hello Aarthi, how are you?
> Looks good,
> Are these cmdlets going to be released soon on the Teams GA or Public Preview version? which version exactly?
> 
> so we can update this later https://docs.microsoft.com/en-us/microsoftteams/teams-powershell-release-notes

@get-itips These will be GA.

**Pull request merged**